Sources/NIOCore/ByteBuffer-views.swift

```diff
@@ -106,6 +113,23 @@ public struct ByteBufferView: RandomAccessCollection {
             return ptr.lastIndex(of: element).map { $0 + self._range.lowerBound }
         })
     }
+
+    @inlinable
+    public func _copyContents(
+      initializing ptr: UnsafeMutableBufferPointer<UInt8>
+    ) -> (Iterator, UnsafeMutableBufferPointer<UInt8>.Index) {
+        precondition(ptr.count >= self.count)
+
+        let bytesToWrite = self.count
+
+        let endIndex = self.withContiguousStorageIfAvailable { ourBytes in
+            ptr.initialize(from: ourBytes).1
+        }
+        precondition(endIndex == bytesToWrite)
+
+        let iterator = self[self.endIndex...self.endIndex].makeIterator()
```

Comment:
what is this iterator suppose to be? Now it is just empty

Reply:
ah no, it is not empty. `self.endIndex...self.endIndex` will result in a range with just `self.endIndex` in it. But this seems wrong...

Reply:
Nope, it's right. [This](https://github.com/apple/swift/blob/ee7e04822c556b481e28dabc4274e54d4ab4f0ae/stdlib/public/core/Sequence.swift#L360-L385) is the customisation point we're implementing. From it we need to return an iterator on _our_ type which contains any elements we couldn't copy into the buffer. Because we can always copy the elements into the buffer, we return the empty iterator.

Reply:
Wouldn't it then be `self.endIndex..<self.endIndex` instead of `self.endIndex...self.endIndex`, in other words a `Range` instead of a `ClosedRange` to create a real empty range? Otherwise the iterator will try to iterate over one element but will fail to do say because it is outside the bounds of `self`.

Reply:
The test currently crash too. If I change it to a `Range` they no longer crash and pass.

Reply:
Yeah good catch, I constructed this incorrectly.